src/Provider/IP2Location/IP2Location.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\IP2Location;
+
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Collection;
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Provider\Provider;
+use Http\Client\HttpClient;
+
+/**
+ * @author William Durand <william.durand1@gmail.com>
+ */
+final class IP2Location extends AbstractHttpProvider implements Provider
+{
+    /**
+     * @var string
+     */
+    const ENDPOINT_URL = 'https://api.ip2location.com/v2/?key=%s&ip=%s&format=json&package=WS9';
```

Comment:
I would set the version number and `package` as constant/variables !
And add documentation about what the `package` is !

Reply:
This is the version the version and package that matched Geocoder fields perfectly. Therefore, it will remain here.

Reply:
Therefore I would clearly state in the README that each request would cost 4 credits (according to [documentation](https://ip2location.com/web-service/ip2location))

Reply:
@jbelien Thanks for your suggestion, we have updated the README.